### PR TITLE
Add nlohmann-json package

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Add nlohmann-json package as it's required by some of our partners.

###### Change Log  

- Updated Source0

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local build of affected package

### CODE REVIEW

File: SPECS/nlohmann-json/nlohmann-json.spec [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Version:        3.6.1
+Release:        2%{?dist}
+License:        MIT
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          System Environment
+URL:            https://github.com/nlohmann/json
+#Source0:       https://github.com/nlohmann/json/archive/v%{version}.tar.gz
+Source0:        %{name}-%{version}.tar.gz
+%global debug_package %{nil}
```
Comment (PawelWMS) [inline SPECS/nlohmann-json/nlohmann-json.spec:12]: I think we can remove that. We'd like to have debug symbols for as many packages as we can.

File: SPECS/nlohmann-json/nlohmann-json.spec [hunk 2]
```diff
@@ -0,0 +1,55 @@
+Summary:        Modern C++11 JSON library
+Name:           nlohmann-json
```
Comment (PawelWMS) [inline SPECS/nlohmann-json/nlohmann-json.spec:2]: Please also update `SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md ` for both packages.

Comment (thcrain-msft) [reply SPECS/nlohmann-json/nlohmann-json.spec:2]: Fixed alongside other comments from this pass, thanks!

**Review by PawelWMS — APPROVED** [review]
:shipit:

**Pull request merged**